Automattic/jetpack
# Publicize: Add new publicize icon toggle component

Fixes #21064

#### Changes proposed in this Pull Request:

* Add new `ConnectionToggle` component that includes service profile picture, username, icon, and toggle.
* Add new `ConnectionIcon` component that includes service profile picture, username, and icon.
* Update Facebook and Twitter social icons to fulfill brand guidelines. This will also change the icons in the Social Previews panel and modal.
* Update Publicize module and endpoint schemas to include `profile_picture`
* Refresh Publicize connections and set a transient to stagger requests

<img width="290" alt="Screen Shot 2021-10-18 at 11 58 23 AM" src="https://user-images.githubusercontent.com/1689238/137766970-c5b612c1-0048-4944-b206-48458e63aeca.png">

#### Jetpack product discussion



p6TEKc-5mX-p2

#### Does this pull request change what data or activity we track or use?



Nope.

#### Testing instructions:






* Enable the Publicize module on a Jetpack site and checkout this branch. 
* Apply this patch to your sandbox and sandbox the API: D68827-code
* Sandbox the Jetpack site with the `JETPACK__SANDBOX_DOMAIN` constant.
* Create/edit a post.
* Open the Jetpack menu in the editor (Jetpack logo in the upper right).
* Verify that all of the connection profile pictures are displaying correctly (you may need to delete the transient, e.g. `jetpack docker wp transient delete jetpack_publicize_connection_refresh_wait`).
* Toggle on and off the connections and verify that they disable or enable the connection when publishing.
* Check that the social icon is grayed out when a connection is toggled off.
* Check that social connections without profile pictures display the social icon by itself.
* Check that the Social Previews panel and modal load the correct social icons for Twitter and Facebook.
* Repeat the steps above for an Atomic site.
* Apply the sync patch from this PR to your sandbox: D66383-code
* Repeat the testing steps for a sandboxed Simple site.
